src/DeploymentProvider/Providers/WebAppDeploymentProvider.ts

```diff
@@ -84,4 +91,50 @@ export class WebAppDeploymentProvider extends BaseWebAppDeploymentProvider {
         console.log('App Service Application URL: ' + this.applicationURL);
         core.setOutput('webapp-url', this.applicationURL);
     }
+
+    private async deleteReleaseZipForLinuxPhpApps(webPackage: string): Promise<void> {
+        
+        // No need to delete release.zip for non-linux apps
+        if (!this.actionParams.isLinux) {
+            return;
+        }
+
+        let isPhpApp = await this.checkIfTheAppIsPhpApp(webPackage);
+
+        // No need to delete release.zip for non-PHP apps
+        if (!isPhpApp) {
+            return;
+        }
+
+        // Delete release.zip if it exists
+
+        try {
+            const releaseZipPath = path.join(webPackage, 'release.zip');
+            await fs.promises.unlink(releaseZipPath);
+            core.debug(`Deleted release.zip`);
+        } catch (error) {
+            core.debug(`Error while deleting release.zip for Linux PHP app: ${error}`);
+        }
+    }
+
+    private async checkIfTheAppIsPhpApp(webPackage: string): Promise<boolean> {
+
+        try {
+            // Check if the web package contains a composer.json file
+            const composerFile = 'composer.json'; 
+            if (fs.existsSync(path.join(webPackage, composerFile))) {
+                return true;
+            }
+
+            // Check if the web package contains a .php file
+            const phpFiles = fs.readdirSync(webPackage).filter(file => file.endsWith('.php'));

```

Comment:
Currently this iterates over all the files, how about we check for first PHP and exit early.
something like following.
```
const files = fs.readdirSync(webPackage);
for (const file of files) {
    if (file.endsWith('.php')) {
        return true;
    }
}
```